hnswlib/space_l2.h

```diff
@@ -208,7 +213,17 @@ namespace hnswlib {
     public:
         L2Space(size_t dim) {
             fstdistfunc_ = L2Sqr;
-        #if defined(USE_SSE) || defined(USE_AVX) || defined(USE_AVX512)
+    #if defined(USE_SSE) || defined(USE_AVX) || defined(USE_AVX512)
+        #if defined(USE_AVX512)
+            if (AVX512Capable())
+                L2SqrSIMD16Ext = L2SqrSIMD16ExtAVX512;
+            else if (AVXCapable())
+                L2SqrSIMD16Ext = L2SqrSIMD16ExtAVX;
+        #elif defined(USE_AVX)
+            if (AVXCapable())
+                L2SqrSIMD16Ext = L2SqrSIMD16ExtAVX;
```

Comment:
This line overwrites the AVX512 selection above, so AVX512 is not used if it is supported. The same thing is for the inner product distance.

Reply:
That leads to AVX512 code never running

Reply:
The logic uses an "else if", so if AVX512 was compiled then it checks at run time to see if it is able to run, if not it will use AVX.  It will only go into the "USE_AVX" code block if it is not compiled on an AVX512 capable machine. This was confirmed with print statements showing it used the AVX512 code block.

Reply:
Ah. Ok. The bug is only in inner product then